packages/react/src/components/ExpressiveModal/ExpressiveModal.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings as ddsSettings } from '@carbon/ibmdotcom-utilities';
+import { settings } from 'carbon-components';
+import { ComposedModal } from 'carbon-components-react';
+import ExpressiveModalCloseBtn from './ExpressiveModalCloseBtn';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+/**
+ * Expressive Modal
+ *
+ * @param {object} props react proptypes
+ * @returns {object} JSX object
+ */
+const ExpressiveModal = ({ open, className, children, onClose, ...props }) => {
+  const [isOpen, setIsOpen] = useState(open);
+  const [prevOpen, setPrevOpen] = useState(open);
+
+  useEffect(manageOpenState);
+
+  return (
+    <ComposedModal
+      onClose={onClose}
+      open={isOpen}
+      data-autoid={`${stablePrefix}--expressive-modal`}
+      className={classNames(`${prefix}--modal--expressive`, className)}
+      {...props}>
+      <ExpressiveModalCloseBtn onClick={closeModal} />
+      {children}
+    </ComposedModal>
+  );
+
+  /**
+   * Close modal
+   */
+  function closeModal() {
+    if (!onClose || onClose() !== false) {
+      setIsOpen(false);
+    }
+  }
+
+  /**
+   * Manage open prop and isOpen state
+   */
+  function manageOpenState() {
+    if (prevOpen !== open) {
+      setIsOpen(open);
```

Comment:
@photodow one thing I neglected to add in the acceptance criteria are the analytics triggered events in this modal. We could probably do this in a separate PR, but that means we'd have to add it post code freeze if we can't sneak this in today.

Reply:
I was assuming that was part of the https://github.com/carbon-design-system/ibm-dotcom-library/issues/391 story. I'm fine with a separate pull request. I don't think code freeze has to be so strict, but it's up to you. At the very least #571 needs to go in first, right?

Reply:
@photodow Yes, #571 definitely has to go in first. I'm fine with getting this in separately.